# NOTEBOOK #01 — Inspección Inicial

**Proyecto Integrador — Minería de Datos I — Tec. Sup. Ciencia de Datos — ITSE**  
**Integrantes:** Daniela Fernandez — Julio Nahuel Gomez  
**Profesor:** Fernando Elias Mubarqui  
**Año:** 2026

**Dataset:** Usuarios de plataforma de streaming  
**Objetivo:** Explorar la estructura general del dataset, identificar tipos de datos, 
valores faltantes, duplicados e inconsistencias, sin realizar ninguna modificación 
sobre los datos originales.

In [25]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [26]:
df = pd.read_json('../data/raw/streaming_users_dirty.csv')
df.head()

,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
0,10000,39,Estándar,805.8,Brasil,Crime,2025-03-04,99
1,10001,37,Estándar,1173.4,Colombia,Crime,2019-04-02,2
2,10002,28,Básico,401.0,Colombia,Crime,2018-04-13,0
3,10003,43,Básico,62.4,Uruguay,Thriller,2021-01-31,0
4,10004,51,Básico,477.8,Perú,Thriller,2020-09-30,1


## ESTRUCTURA GENERAL DEL DATASET

Verificamos cuántas filas y columnas tiene el dataset y los nombres de cada columna.

In [27]:
print(f"Filas: {df.shape[0]}")
print(f"Columnas: {df.shape[1]}")
print(f"\nNombre de las columnas: {list(df.columns)}")

Filas: 8160
Columnas: 8

Nombre de las columnas: ['user_id', 'age', 'subscription_plan', 'monthly_watch_time_mins', 'country', 'favorite_genre', 'last_login_date', 'customer_support_tickets']


In [28]:
print("Tipos de datos:")
print(df.dtypes)

Tipos de datos:
user_id                       int64
age                           int64
subscription_plan               str
monthly_watch_time_mins     float64
country                         str
favorite_genre                  str
last_login_date                 str
customer_support_tickets      int64
dtype: object


Observamos que:
- `user_id` → int64 (número entero)
- `age` → int64 (número entero)
- `subscription_plan` → object (texto)
- `monthly_watch_time_mins` → float64 (decimal)
- `country` → object (texto)
- `favorite_genre` → object (texto)
- `last_login_date` → object (texto) — debería ser de tipo fecha
- `customer_support_tickets` → int64 (número entero)

In [29]:
print("Valores faltantes por columna:")
print(df.isnull().sum())
print(f"\nTotal de valores faltantes: {df.isnull().sum().sum()}")

Valores faltantes por columna:
user_id                       0
age                           0
subscription_plan             0
monthly_watch_time_mins     193
country                       0
favorite_genre              240
last_login_date             320
customer_support_tickets      0
dtype: int64

Total de valores faltantes: 753


Encontramos 753 valores faltantes en total distribuidos así:
- `monthly_watch_time_mins` → 193 nulos
- `favorite_genre` → 240 nulos
- `last_login_date` → 320 nulos

In [30]:
duplicados = df.duplicated().sum()
print(f"Registros duplicados: {duplicados}")

Registros duplicados: 126


Hay 126 registros duplicados. Eso significa que 126 filas aparecen más de una vez con exactamente los mismos datos.

In [31]:
print("Valores únicos en subscription_plan:")
print(df['subscription_plan'].unique())

print("\nValores únicos en country:")
print(df['country'].unique())

print("\nValores únicos en favorite_genre:")
print(df['favorite_genre'].unique())

Valores únicos en subscription_plan:
<ArrowStringArray>
[ 'Estándar',    'Básico',   'Premium',       'Std',  'estandar',    'basico',
    'básico',  'Premium ',   'premium',   'Premiun',    'BASICO',  'STANDARD',
     'Basic', 'Estándar ',   'PREMIUM']
Length: 15, dtype: str

Valores únicos en country:
<ArrowStringArray>
[    'Brasil',   'Colombia',    'Uruguay',       'Perú',      'Chile',
  'Argentina',     'México',     'Brazil',     'brasil',     'méxico',
      'chile',    'uruguay',        'MEX',        'ARG',   'colombia',
        'COL',     'Mexico',        'URY',     'Chile ',       'Peru',
       'perú',  'argentina',        'PER', 'Argentina ',        'BRA',
        'CHL']
Length: 26, dtype: str

Valores únicos en favorite_genre:
<ArrowStringArray>
[      'Crime',    'Thriller',       'Drama',      'Acción',     'Romance',
     'Comedia',  'Documental',      'ACCIÓN',           nan,       'CRIME',
    'Comedia ',      'comedy',       'DRAMA',    'THRILLER', 'Documentary',
 

Observamos que:
- `subscription_plan` tiene 15 variantes para solo 3 planes reales: Básico, Estándar y Premium escritos con mayúsculas, minúsculas, acentos, abreviaciones y errores tipográficos.
- `country` tiene países escritos con nombres completos, abreviados, en minúsculas y con espacios extra.
- `favorite_genre` tiene géneros en español e inglés mezclados, con mayúsculas, minúsculas y errores tipográficos.

In [32]:
df[['age', 'monthly_watch_time_mins', 'customer_support_tickets']].describe()

,age,monthly_watch_time_mins,customer_support_tickets
count,8160.000000,7967.000000,8160.000000
mean,34.096814,1107.346153,1.800980
std,14.511304,5310.442604,11.334969
min,-5.000000,-120.000000,-1.000000
25%,25.000000,489.200000,0.000000
50%,33.000000,757.400000,1.000000
75%,42.000000,1045.700000,1.000000
max,150.000000,99999.000000,150.000000


Observamos valores imposibles en las tres columnas numéricas:
- `age` → mínimo -5 y máximo 150, ambos imposibles para una edad real.
- `monthly_watch_time_mins` → mínimo -120 (tiempo negativo) y máximo 99999 (más de 69 días seguidos, claramente un error).
- `customer_support_tickets` → mínimo -1 (tickets negativos) y máximo 150 (valor extremadamente alto).

## OBSERVACIONES FINALES

A partir de la inspección identificamos los siguientes problemas en el dataset:

- **Dimensiones:** 8.160 filas y 8 columnas.
- **Valores faltantes:** 753 en total. Columnas afectadas: `monthly_watch_time_mins` (193), `favorite_genre` (240) y `last_login_date` (320).
- **Duplicados:** 126 registros duplicados.
- **Tipo de dato incorrecto:** `last_login_date` figura como texto cuando debería ser fecha.
- **Inconsistencias categóricas:** `subscription_plan`, `country` y `favorite_genre` presentan múltiples variantes para un mismo valor.
- **Valores imposibles:** edades negativas y mayores a 120, tiempos negativos y extremadamente altos, tickets negativos.

Todos estos problemas serán tratados en el Notebook 02.